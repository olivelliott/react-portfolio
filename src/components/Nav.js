import React from "react";

function Nav({ currentPage, handlePageChange }) {

  const determineActiveClass = (pageName) =>
    `nav-link ${currentPage === pageName ? "active" : ""}`;

  const pageNames = ["About", "Portfolio", "Contact"];

  return (
    <ul className="nav nav-tabs">
      {pageNames.map((name) => {
        return (
          <li className="nav-item" key={name}>
            <a
              href={`#${name}`}
              onClick={() => handlePageChange(name)}
              className={determineActiveClass(currentPage, name)}
            >
              {name}
            </a>
          </li>
        );
      })}
    </ul>
  );
}

export default Nav;
