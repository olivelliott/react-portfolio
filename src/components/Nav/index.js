import React from 'react';
// import About from '../About'

function Nav(props) {
    const {
        categories = [],
        setCurrentCategory,
        currentCategory
    } = props;

    return (
        <header className='flex-row px-1'>
            <h2>
                <a data-testid='link' href='/'>
                    OLIVE
                </a>
            </h2>
            <nav>
                <ul className='flex-row'>
                    {categories.map((category) => (
                        <li className={`mx-1 ${
                            currentCategory.name === category.name && 'navActive'
                            }`} key={category.name}>
                            <span 
                                onClick={() => {
                                    setCurrentCategory(category)
                                }} >
                                {category.name}
                            </span>
                        </li>
                    ))}
                </ul>
            </nav>
        </header>
    )

}

export default Nav;