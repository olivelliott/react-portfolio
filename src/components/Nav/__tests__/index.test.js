import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

const categories = [
    { name: 'About' }
]

afterEach(cleanup);

describe('Nav component', () => {
    it('renders without crashing', () => {
        render(<Nav />);
    });

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Nav />);
        expect(asFragment()).toMatchSnapshot();
    })
})

describe('links are visible', () => {
    it('inserts text into the links', () => {
        const { getByTestId } = render(<Nav />);
        expect(getByTestId('link')).toHaveTextContent('OLIVE');
        expect(getByTestId('about')).toHaveTextContent('About');
    })
})
