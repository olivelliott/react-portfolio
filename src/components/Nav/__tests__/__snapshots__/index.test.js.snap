// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        OLIVE
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About
          </a>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Portfolio
          </span>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-2"
        >
          <span>
            Resume
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
